llvm/lib/CAS/OnDiskKeyValueDB.cpp

```diff
@@ -55,13 +66,19 @@ OnDiskKeyValueDB::open(StringRef Path, StringRef HashName, unsigned KeySize,
   constexpr uint64_t MB = 1024ull * 1024ull;
   constexpr uint64_t GB = 1024ull * 1024ull * 1024ull;
 
+  uint64_t MaxFileSize = GB;
+  uint64_t CustomSize = 0;
+  if (Error E = getConfingFromEnv("LLVM_CAS_MAX_MAPPING_SIZE", CustomSize))
```

Comment:
This is repeated, I'd suggest to put a `getConfiguredMaxMappingSize` function in a `OnDiskCommon.h` header and include it in both files.